frontend/components/tree/js/tree.service.ts

```diff
@@ -936,56 +993,105 @@ export class TreeService {
+
+		this.ViewerService.clearHighlights();
+		this.DocsService.closeDocs();
+
 		if (this.currentSelectedNodes) {
-			this.currentSelectedNodes.forEach((selectedNode) => {
-				selectedNode.selected = false;
-			});
+			for (let i = 0; i < this.currentSelectedNodes.length; i++) {
+				this.currentSelectedNodes[i].selected = false;
+			}
 		}
+
 		this.currentSelectedNodes = [];
 	}
 
 	/**
 	 * Set selection status of node.
-	 * @param node		Node to set.
-	 * @param isSelected	Is selected.
+	 * @param node	Node to set selection status of
+	 * @param select whether the node should be selected or not
 	 */
-	public setNodeSelection(node: any, isSelected: boolean) {
-		if (node.selected !== isSelected) {
-			const nodeIndex = this.currentSelectedNodes.indexOf(node);
-
-			if (isSelected) {
-				if (-1 === nodeIndex) {
-					node.selected = true;
-					this.currentSelectedNodes.push(node);
-				}
-			} else {
-				if (nodeIndex > -1) {
-					this.currentSelectedNodes[nodeIndex].selected = false;
-					this.currentSelectedNodes.splice(nodeIndex, 1);
-				}
+	public setNodeSelection(node: any, select: boolean) {
+
+		// If node
+		if (node.selected === select) {
+			return;
+		}
+
+		const nodeIndex = this.currentSelectedNodes.indexOf(node);
+
+		if (select) {
+			if (nodeIndex === -1) {
+				node.selected = true;
+				this.currentSelectedNodes.push(node);
+			}
+		} else {
+			if (nodeIndex > -1) {
+				this.currentSelectedNodes[nodeIndex].selected = false;
+				this.currentSelectedNodes.splice(nodeIndex, 1);
 			}
+		}
+	}
 
-			return this.ready.promise.then(() => {
-				const currentSelectedMap = {};
-				this.currentSelectedNodes.forEach((n) => {
-					this.traverseNodeAndPushId(n, currentSelectedMap, this.treeMap.idToMeshes);
-				});
+	public getMeshHighlights(nodes) {
+		return this.ready.promise.then(() => {
+			const currentSelectedMap = this.getMeshMapFromNodes(nodes, this.treeMap.idToMeshes);
+			return currentSelectedMap;
+		});
+	}
 
-				this.setHighlightMap(currentSelectedMap);
-				return currentSelectedMap;
-			});
+	/**
+	 * Return a map of currently selected meshes
+	 */
+	public getCurrentMeshHighlights() {
+		return this.getMeshHighlights(this.currentSelectedNodes.concat());
+	}
+
+	/**
+	 * Deselect a nodes in the tree.
+	 * @param nodes	Node to select.
+	 */
+	public deselectNodes(nodes: any[]) {
+
+		for (let i = 0; i < nodes.length; i++) {
+			const node = nodes[i];
+			this.setNodeSelection(node, false);
 		}
+
+		return this.highlightNodes(nodes, true, undefined, false);
 	}
 
 	/**
-	 * Select a node in the tree.
-	 * @param node	Node to select.
+	 * Select nodes in the tree.
+	 * @param nodes	Nodes to select.
 	 * @param multi	Is multi select enabled.
 	 */
-	public selectNode(node: any, multi: boolean, final: boolean) {
+	public selectNodes(
+		nodes: any[], multi: boolean, additive: boolean, colour: number[],
+		forceReHighlight: boolean,
+	) {
+
+		forceReHighlight = (forceReHighlight === undefined) ? true : false;
```

Comment:
default behaviour should be false. This line doesn't need to exist?

Reply:
This is sorted now